index.js

```diff
@@ -171,5 +171,22 @@ function validateForm(e) {
     e.stopPropagation();
   }
 }
-const form3 = document.forms.form2;
-form3.addEventListener('submit', validateForm);
+const form2 = document.forms.form1;
+form2.addEventListener('submit', validateForm);
+
+const name1 = document.getElementById('name');
+const email1 = document.getElementById('email');
+const textarea1 = document.getElementById('textarea');
+
+form2.addEventListener('input', () => {
+  const formData = {
+    name: name1.value,
+    email: email1.value,
+    textareamessage: textarea1.value,
+  };
+  localStorage.setItem('user-data', JSON.stringify(formData));
+});
+const getDataFromLocalStorage = JSON.parse(localStorage.getItem('user-data'));
+name1.value = getDataFromLocalStorage.name;
+email1.value = getDataFromLocalStorage.email;
+textarea1.value = getDataFromLocalStorage.textareamessage;
```

Comment:
- [ ] Good job implementing the features to preserve data :clap:  however, please notice you are not checking for null value initially that's why there is an error in the console :point_down: 
![image](https://user-images.githubusercontent.com/97521557/216752297-656ef76b-3aa6-433a-875a-4e7977fd9851.png)

please fix this by implementing a conditional check like below  :point_down: 
```
if(getDataFromLocalStorage){
fname.value =  getDataFromLocalStorage.name;
emailval.value =  getDataFromLocalStorage.email;
messages.value =  getDataFromLocalStorage.textareamessage;
}
```